lib/search_song_page.dart

```diff
@@ -53,6 +53,83 @@ class _MySearchSongPageState extends State<MySearchSongPage> {
     });
   }
 
+  void _updateSearchResults(String searchText) {
+    searchResults = [];
+    String lastSongSeen = '';
+    for (var element in allSearchVerses) {
+      // Continue with next verse if search text is not found in this one.
+      if (!(element.text.toLowerCase().contains(searchText.toLowerCase()))) {
+        continue;
+      }
+      // Add the song title as a header for its found verses.
+      if (lastSongSeen != element.songKey) {
+        lastSongSeen = element.songKey;
+        searchResults.add(ListTile(
+          title: Text(
+            '${element.songKey}. ${widget.songs[element.songKey]['title']}',
+            maxLines: 1,
+            overflow: TextOverflow.ellipsis,
+          ),
+          // Song titles should have no left padding.
+          contentPadding: const EdgeInsets.only(
+            left: 3,
+            right: 3,
+          ),
+        ));
+      }
+
+      // Highlight search text by making it bold.
+      final matchPosition =
+          element.text.toLowerCase().indexOf(searchText.toLowerCase());
+      List<TextSpan> titleSpans = [
+        TextSpan(
+          text: element.text.substring(0, matchPosition),
+        ),
+        TextSpan(
+          text: element.text
+              .substring(matchPosition, matchPosition + searchText.length),
+          style: const TextStyle(
+            // This is the boldest possible choice; changing the color is not
+            // really an option as both dark and light modes should be
+            // supported.
```

Comment:
Theme.of(context) has a color that is set to look good in both dark and light mode. Maybe use it as a background color for the text span?
Or: cardColor, which is not that much different from the background, but stands out a little.

Reply:
Köszi!